jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/EndEventHandler.java

```diff
@@ -202,9 +204,43 @@ public void handleMessageNode(final Node node, final Element element, final Stri
     protected void readEndDataInputAssociation(org.w3c.dom.Node xmlNode, EndNode endNode) {
         // sourceRef
         org.w3c.dom.Node subNode = xmlNode.getFirstChild();
-        String eventVariable = subNode.getTextContent();
-        if (eventVariable != null && eventVariable.trim().length() > 0) {
-            endNode.setMetaData("MappingVariable", eventVariable);
+        if ("sourceRef".equals(subNode.getNodeName())) {
+            String eventVariable = subNode.getTextContent();
+            if (eventVariable != null && eventVariable.trim().length() > 0) {
+                if (dataInputs.containsKey(eventVariable)) {
+                    eventVariable = dataInputs.get(eventVariable);
+                }
+                
+                endNode.setMetaData("MappingVariable", eventVariable);
+            }
+        } else {
+            // targetRef
+            String to = subNode.getTextContent();
```

Comment:
It seems that this variable is not needed at all, although I would leave here // targetRef comment.